keys/employees-keys/README.md

```diff
@@ -3,172 +3,117 @@
+  - [What to do when private key was lost or
+    compromised?](#what-to-do-when-private-key was-lost-or-compromised)
+
 ## Adding key to the repository
 
 To add new key to this repository you should:
 
 1. [Create new GPG key](#how-to-create-a-new-gpg-key)
-2. [Send key for signing](#sending-key-for-signing)
+2. [Encrypt revocation cert](#encrypt-revocation-cert)
+3. [Send key for signing and revocation
+   cert](#sending-key-for-signing-and-revocation-cert)
 
 Then 3mdeb Team Leader or Management should
 
-1. [3mdeb Team Leader/Management: Identity confirmation](#identity-confirmation)
+1. [3mdeb Team Leader/Management: Identity
+   confirmation](#identity-confirmation)
 2. [3mdeb Team Leader/Management: Key signing](#key-signing)
 
 Finally you should configure your git and provide signed key through pull
 request:
 
 1. [Git configuration](#git-configuration)
 2. [Create Pull Request with signed key](#create-a-pull-request-with-key)
+3. [Upload to keys.opengpg.org](#upload-to-keys-openpgp-org)
 
 If you have questions, please check [FAQ](#FAQ) section. Maybe the answer is
 already there. If not, feel free to submit an issue to the repository.
 
-### How to create a new GPG key
+### Create a new GPG key
 
 The following assumes that you have `gpg` installed.  The interaction is broken
 into sections that correspond to user queries.
 
-```
-$ gpg --gen-key
-```
-
-For GPG version 2, you should use `gpg --full-generate-key`. Please note that
-depending on the version of GPG you use, your output may look different.
-
-This is what you'll see if `gpg` is configured for the first time:
-
-```
-gpg (GnuPG) 2.2.27; Copyright (C) 2021 Free Software Foundation, Inc.
-This is free software: you are free to change and redistribute it.
-There is NO WARRANTY, to the extent permitted by law.
-```
-
-Pick "RSA and RSA" key:
-
-```
-Please select what kind of key you want:
-   (1) RSA and RSA (default)
-   (2) DSA and Elgamal
-   (3) DSA (sign only)
-   (4) RSA (sign only)
-Your selection? 1
-```
-
-Use 4096 bits for key size:
+Clone repository:
 
-```
-RSA keys may be between 1024 and 4096 bits long.
-@What keysize do you want? (2048) 4096
-Requested keysize is 4096 bits
+```shell
+git clone https://github.com/3mdeb/3mdeb-secpack.git
 ```
 
-Use a 2 years expiry key and then confirm your choice:
+Change directory:
 
-```
-Please specify how long the key should be valid.
-         0 = key does not expire
-      <n>  = key expires in n days
-      <n>w = key expires in n weeks
-      <n>m = key expires in n months
-      <n>y = key expires in n years
-Key is valid for? (0) 2y
-Key expires at Fri 07 Apr 2023 11:40:33 AM CEST
-Is this correct? (y/N) y
+```shell
+cd 3mdeb-secpack/scripts
 ```
 
-Enter your real name:
+Generate key:
 
-```
-GnuPG needs to construct a user ID to identify your key.
-
-@Real name: Your Name
+```shell
+./gen-new-employee-key.sh "Your Name" "your.name@3mdeb.com"
```

Comment:
For me, it ends with an Error. Here is a debug output from script:

```
./gen-new-employee-key.sh "Your Name" "your.name@3mdeb.com"
+ '[' 2 -ne 2 ']'
+ KEY_NAME='Your Name'
+ KEY_EMAIL=your.name@3mdeb.com
+ KEY_COMMENT='Employee Cert Key'
+ KEY_FILE=your.name@3mdeb.com.asc
+ REVOCATION_FILE=your.name@3mdeb.com-revocation.asc
++ cat
+ BATCH_FILE_CONTENT='%echo Generating a primary certification key and subkeys for signing and encryption
Key-Type: RSA
Key-Length: 4096
Key-Usage: cert
Expire-Date: 0
Subkey-Type: RSA
Subkey-Length: 4096
Subkey-Usage: sign
Name-Real: Your Name
Name-Comment: Employee Cert Key
Name-Email: your.name@3mdeb.com
%no-protection
%commit
%echo Key generation complete'
+ echo '%echo Generating a primary certification key and subkeys for signing and encryption
Key-Type: RSA
Key-Length: 4096
Key-Usage: cert
Expire-Date: 0
Subkey-Type: RSA
Subkey-Length: 4096
Subkey-Usage: sign
Name-Real: Your Name
Name-Comment: Employee Cert Key
Name-Email: your.name@3mdeb.com
%no-protection
%commit
%echo Key generation complete'
++ gpg --batch --generate-key temp_batch_file.txt
++ grep 'gpg: key'
++ awk '{print $3}'
+ KEY_ID=
+ '[' -z '' ']'
+ echo 'Error: Could not retrieve KEY_ID.'
Error: Could not retrieve KEY_ID.
+ exit 1
```

Reply:
```
gpg --version
gpg (GnuPG) 2.4.0
libgcrypt 1.10.2-unknown
Copyright (C) 2021 Free Software Foundation, Inc.
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /home/macpijan/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```

Reply:
@pietrushnic  Can you show the debug output on your side, so we can more easily find what the problem is?

Reply:
@macpijan I assume this is GPG version issue:

```
gpg (GnuPG) 2.2.40
libgcrypt 1.10.1
Copyright (C) 2022 g10 Code GmbH
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /home/user/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```

Output is:

```
+ '[' 2 -ne 2 ']'
+ KEY_NAME='Your Name'
+ KEY_EMAIL=your.name@3mdeb.com
+ KEY_COMMENT='Employee Cert Key'
+ KEY_FILE=your.name@3mdeb.com.asc
++ cat
+ BATCH_FILE_CONTENT='%echo Generating a primary certification key and subkeys for signing and encryption
Key-Type: RSA
Key-Length: 4096
Key-Usage: cert
Expire-Date: 0
Subkey-Type: RSA
Subkey-Length: 4096
Subkey-Usage: sign
Name-Real: Your Name
Name-Comment: Employee Cert Key
Name-Email: your.name@3mdeb.com
%no-protection
%commit
%echo Key generation complete'
+ echo '%echo Generating a primary certification key and subkeys for signing and encryption
Key-Type: RSA
Key-Length: 4096
Key-Usage: cert
Expire-Date: 0
Subkey-Type: RSA
Subkey-Length: 4096
Subkey-Usage: sign
Name-Real: Your Name
Name-Comment: Employee Cert Key
Name-Email: your.name@3mdeb.com
%no-protection
%commit
%echo Key generation complete'
++ gpg --batch --generate-key temp_batch_file.txt
++ grep 'gpg: key'
++ awk '{print $3}'
+ KEY_ID=generation
+ '[' -z generation ']'
+ echo 'Modifying key: generation'
Modifying key: generation
+ gpg --command-fd 0 --edit-key generation
gpg (GnuPG) 2.2.40; Copyright (C) 2022 g10 Code GmbH
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

gpg: key "generation" not found: No public key
+ '[' 2 -ne 0 ']'
+ echo 'Error: Cannot change expiration date for signing subkey.'
Error: Cannot change expiration date for signing subkey.
+ exit 1
```

Let me tinker with the script more, it looks like it is not robust enough.

Reply:
Ok, please ping me if retest is needed @pietrushnic

Reply:
@macpijan please check bbc5c62d3b0f (sample debug run in commit message).

Reply:
@macpijan can you try it again?

Reply:
@macpijan ping?

Reply:
It works for me now.